pages/embedded-portal.tsx

```diff
@@ -264,16 +274,83 @@ export const getServerSideProps: GetServerSideProps = async (context) => {
     },
   });
 
+  if (!existingSpace) {
+    console.log("No space");
+    return {
+      props: {
+        environmentToken,
+        lastSyncedAt: lastSync
+          ? DateTime.fromJSDate(lastSync.createdAt).toFormat(
+              "MM/dd/yy hh:mm:ss a"
+            )
+          : "",
+        existingSpace,
+        userId: token.sub,
+      },
+    };
+  }
+
+  let spaceData = await getSpace({
+    workflow: WorkflowType.Embedded,
+    spaceId: (existingSpace?.flatfileData as unknown as FlatfileSpaceData).id,
+  });
+
+  console.log("spaceData", spaceData);
+
+  let workbook = await getWorkbook({
+    workflow: WorkflowType.Embedded,
+    workbookId: spaceData?.primaryWorkbookId!,
+  });
+
+  console.log("workbook", workbook);
+
+  const timeInFive = new Date();
+  timeInFive.setSeconds(timeInFive.getSeconds() + 5);
+
+  while (!workbook || new Date() > timeInFive) {
+    spaceData = await getSpace({
+      workflow: WorkflowType.Embedded,
+      spaceId: (existingSpace?.flatfileData as unknown as FlatfileSpaceData).id,
+    });
+
+    workbook = await getWorkbook({
+      workflow: WorkflowType.Embedded,
+      workbookId: spaceData?.primaryWorkbookId!,
+    });
+  }
+
+  if (!workbook) {
+    console.log("Unable to get workbook");
+    throw "Unable to get workbook";
```

Comment:
What happens if this throws? Should we redirect with an error message instead?

Reply:
edited to redirect